modules/finance/src/main/java/com/opengamma/strata/finance/rate/future/DeliverableSwapFuture.java

```diff
@@ -0,0 +1,704 @@
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.value.Rounding;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.id.LinkResolutionException;
+import com.opengamma.strata.collect.id.LinkResolver;
+import com.opengamma.strata.collect.id.Resolvable;
+import com.opengamma.strata.collect.id.StandardId;
+import com.opengamma.strata.finance.Product;
+import com.opengamma.strata.finance.Security;
+import com.opengamma.strata.finance.SecurityLink;
+import com.opengamma.strata.finance.rate.swap.ExpandedSwapLeg;
+import com.opengamma.strata.finance.rate.swap.NotionalExchange;
+import com.opengamma.strata.finance.rate.swap.NotionalPaymentPeriod;
+import com.opengamma.strata.finance.rate.swap.PaymentEvent;
+import com.opengamma.strata.finance.rate.swap.PaymentPeriod;
+import com.opengamma.strata.finance.rate.swap.Swap;
+import com.opengamma.strata.finance.rate.swap.SwapLeg;
+
+/**
+ * A deliverable swap futures contract.
+ * <p>
+ * A deliverable swap future is a financial instrument that physically settles an interest rate swap on a future date. 
+ * The futures product is margined on a daily basis. 
+ * This class represents the structure of a single futures contract.
+ */
+@BeanDefinition
+public final class DeliverableSwapFuture
+    implements Product, Resolvable<DeliverableSwapFuture>, ImmutableBean, Serializable {
+
+  /**
+   * The link to the underlying swap.
+   * <p>
+   * This property returns a link to the security via a {@link StandardId}.
+   * See {@link #getUnderlying()} and {@link SecurityLink} for more details.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final SecurityLink<Swap> underlyingLink;
+  /**
+   * The delivery date. 
+   * <p>
+   * The underlying swap is delivered on this date.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LocalDate deliveryDate;
+  /**
+   * The last date of trading.
+   * <p>
+   * This date must be before the delivery date of the underlying swap. 
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LocalDate lastTradeDate;
+  /**
+   * The notional of the futures. 
+   * <p>
+   * This is also called face value or contract value.
+   */
+  @PropertyDefinition(validate = "ArgChecker.notNegative")
+  private final double notional;
+  /**
+   * The definition of how to round the option price, defaulted to no rounding.
+   * <p>
+   * The price is represented in decimal form, not percentage form.
+   * As such, the decimal places expressed by the rounding refers to this decimal form.
+   * For example, the common market price of 99.7125 is represented as 0.997125 which
+   * has 6 decimal places.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final Rounding rounding;
```

Comment:
Comment refers to option price. May be worth leaving this out for now, as not sure it is used.